### PR TITLE
Improve copy found lines to clipboard

### PR DESCRIPTION
**Pull request by andreas-jonsson**
https://sourceforge.net/p/notepad-plus/discussion/331753/thread/3a99fa53/

In the "find results" window it's possible to copy the results to the clipboard, but it always copies exactly what's in the window. This includes line info and file names. Sometimes you just want the actual results.

I have added a new option which does this. I have retained the old option (but it is renamed) since I suppose someone might want it.

E.g if the results look like this:

```
Search "toclipboard" (3 hits in 1 file)
  C:\Users\Andreas\Desktop\notepad-plus-plus\PowerEditor\src\ScitillaComponent\FindReplaceDlg.cpp (3 hits)
	Line 2538: 	const generic_string toClipboard = stringJoin(lines, TEXT("\r\n"));
	Line 2539: 	if (!toClipboard.empty())
	Line 2541: 		if (!str2Clipboard(toClipboard.c_str(), _hSelf))
```

Then by right-clicking on the file name and selecting the new option "Copy (just search results)", you get this in the clipboard:

```
	const generic_string toClipboard = stringJoin(lines, TEXT("\r\n"));
	if (!toClipboard.empty())
		if (!str2Clipboard(toClipboard.c_str(), _hSelf))
```

It is also possible to click on each line and copy just that result.

### CODE REVIEW

File: PowerEditor/src/MISC/Common/Common.cpp [hunk 1]
```diff
@@ -747,4 +747,37 @@ generic_string stringJoin(const std::vector<generic_string> &strings, generic_st
...
+bool str2Clipboard(const TCHAR *str2cpy, HWND hwnd)
+{
+	if (!str2cpy)
+		return false;
+
+	int len2Allocate = lstrlen(str2cpy) + 1;
+	len2Allocate *= sizeof(TCHAR);
+	unsigned int cilpboardFormat = CF_TEXT;
+
+	cilpboardFormat = CF_UNICODETEXT;
```
Comment (ariccio) [inline PowerEditor/src/MISC/Common/Common.cpp]: I noticed that you dropped the `#ifdef UNICODE`/`#endif`....did you mean:

```
unsigned int clipboardFormat = CF_TEXT;
clipboardFormat |= CF_UNICODETEXT;
```

or:

```
clipboardFormat = CF_UNICODETEXT;
```
...or something else?

File: PowerEditor/src/MISC/Common/Common.cpp [hunk 2]
```diff
@@ -747,4 +747,37 @@ generic_string stringJoin(const std::vector<generic_string> &strings, generic_st
...
+	}
+
+	if (!::OpenClipboard(hwnd)) //_pPublicInterface->getHSelf()))
+		return false;
+
+	::EmptyClipboard();
+
+	// Lock the handle and copy the text to the buffer.
+	TCHAR *pStr = (TCHAR *)::GlobalLock(hglbCopy);
+	lstrcpy(pStr, str2cpy);
```
Comment (ariccio) [inline PowerEditor/src/MISC/Common/Common.cpp]: If you're refactoring this, maybe you ought to change `lstrlen` to `_tcslen`, and `lstrcpy(pStr, str2cpy)` to `_tcscpy_s(pStr, (len2Allocate/sizeof(TCHAR)), str2cpy)`?

File: PowerEditor/src/MISC/Common/Common.cpp [hunk 3]
```diff
@@ -747,4 +747,37 @@ generic_string stringJoin(const std::vector<generic_string> &strings, generic_st
...
+	::GlobalUnlock(hglbCopy);
+
+	// Place the handle on the clipboard.
+	::SetClipboardData(cilpboardFormat, hglbCopy);
```
Comment (ariccio) [inline PowerEditor/src/MISC/Common/Common.cpp]: Returning false on all the error conditions, except....actually setting the clipboard data? I guess that's ok, but then why bother checking at 2541?

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 4]
```diff
@@ -2477,6 +2477,75 @@ void Finder::openAll()
...
+	}
+	else
+	{
+		// Plus 2 in order to deal with ": ".
+		return s.substr(2 + firstColon);
+	}
+}
+
+void Finder::copy()
+{
```
Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: Is anything in `Finder` being modified? If not, then maybe this could be const?

**Comment by ariccio** [comment]
Nice work.

**Comment by andreas-jonsson** [comment]
Some methods can be marked as const, but copy() can't (or maybe it can, but infects some other methods so I didn't bother).

The error handling in str2clipboard has been fixed.

**Comment by andreas-jonsson** [comment]
Hmm... Maybe I messed the history on this since I wanted the changes from PR 76 (fix debug build) in this one as well.

**Comment by donho** [comment]
Nice patch. But then, I don't see the reason to keep "Copy(with formatting)".
Could you remove "Copy(with formatting)" command and its code and make only one command "Copy"?
Please close this PR and create a new one because of conflict.

**Comment by andreas-jonsson** [comment]
@donho See #150.

**Pull request closed without merging**